api/include/mfxstructures.h

```diff
@@ -1689,11 +1716,41 @@ typedef struct {
     union {
         mfxU8  *QP;         // Block QP value. Valid when Mode = MFX_MBQP_MODE_QP_VALUE
         mfxI8  *DeltaQP;    // For block i: QP[i] = BrcQP[i] + DeltaQP[i]. Valid when Mode = MFX_MBQP_MODE_QP_DELTA
+#if (MFX_VERSION >= 1034)
+        mfxQPandMode *QPmode; // Block-granularity modes when MFX_MBQP_MODE_QP_ADAPTIVE is set
+#endif
         mfxU64 reserved2;
     };
 } mfxExtMBQP;
 MFX_PACK_END()
 
+#if (MFX_VERSION >= 1034)
+MFX_PACK_BEGIN_USUAL_STRUCT()
+typedef struct {
+    mfxExtBuffer    Header; /* Extension buffer header. Header.BufferId must be equal to MFX_EXTBUFF_INSERT_HEADERS. */
+    mfxU16          SPS;      /* tri-state option to insert SPS */
+    mfxU16          PPS;      /* tri-state option to insert PPS */
+    mfxU16          reserved[8];
+} mfxExtInsertHeaders;
+MFX_PACK_END()
+
+MFX_PACK_BEGIN_USUAL_STRUCT()
+typedef struct {
+    mfxExtBuffer    Header; /* Extension buffer header. Header.BufferId must be equal to MFX_EXTBUFF_ENCODER_IPCM_AREA. */
+    mfxU16          reserve1[10];
+
+    mfxU16          NumArea;  /* Number of Area's */
+    struct {
+        mfxU32      Left; /* Left Area's coordinate. */
+        mfxU32      Top;  /* Top  Area's coordinate. */
+        mfxU32      Right; /* Right Area's coordinate. */
+        mfxU32      Bottom; /* Bottom Area's coordinate. */
+
+        mfxU16      reserved2[8];
+    } Area[64];
```

Comment:
why not dynamic array?

Reply:
I'd suggest to ask in [oneVPL spec](https://spec.oneapi.com/versions/latest/elements/oneVPL/source/VPL_intel_media_sdk.html), @varistar would you comment?

Reply:
Yep, makes sense to add dynamic allocation especially if we have Num above.